src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -140,19 +140,24 @@ Copyright (c) .NET Foundation. All rights reserved.
         <RunArguments Condition="'$(RunArguments)' == '' and '$(StartArguments)' != ''">$(_NetCoreRunArguments) $(StartArguments)</RunArguments>
         <RunArguments Condition="'$(RunArguments)' == ''">$(_NetCoreRunArguments)</RunArguments>
 
-        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(OutDir)</RunWorkingDirectory>
+        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(TargetDir)</RunWorkingDirectory>
       </PropertyGroup>
     </When>
     
     <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(OutputType)' == 'Exe'">
       <PropertyGroup>
         <RunCommand Condition="'$(RunCommand)' == ''">$(TargetPath)</RunCommand>
         <RunArguments Condition="'$(RunArguments)' == ''">$(StartArguments)</RunArguments>
-        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(OutDir)</RunWorkingDirectory>
+        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(TargetDir)</RunWorkingDirectory>
       </PropertyGroup>
     </When>
   </Choose>
 
+  <PropertyGroup>
+    <!-- Ensure $(RunWorkingDirectory) is a full path -->
+    <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' != '' and '$([System.IO.Path]::IsPathRooted($(RunWorkingDirectory)))' != 'true'">$([System.IO.Path]::GetFullPath('$(MSBuildProjectDirectory)/$(RunWorkingDirectory)'))</RunWorkingDirectory>
```

Comment:
Is the extra / needed? Isn't MSBuildProjectDirectory guaranteed to have trailing slash? I've found it very good to keep with the convention of having trailing slashes in properties and concatenating them together without slashes. This limits the risk of setting something to near the root of the file system due to an error where a property is empty.